crates/solver-service/src/apis/order.rs

```diff
@@ -0,0 +1,138 @@
+	_solver: &SolverEngine,
+) -> Result<GetOrderResponse, GetOrderError> {
+	info!("Retrieving order with ID: {}", id);
+
+	let order = process_order_request(&id, _solver).await?;
+
+	Ok(GetOrderResponse { order })
+}
+
+/// Processes an order retrieval request.
+async fn process_order_request(
+	order_id: &str,
+	solver: &SolverEngine,
+) -> Result<OrderResponse, GetOrderError> {
+	// Validate order ID format
+	validate_order_id(order_id)?;
+
+	// Try to retrieve the order from storage
+	match solver
+		.storage()
+		.retrieve::<solver_types::Order>("orders", order_id)
+		.await
+	{
+		Ok(order) => {
+			// Order found in storage, convert to OrderResponse
+			convert_order_to_response(order).await
+		}
+		Err(solver_storage::StorageError::NotFound) => {
+			// Order not found in storage
+			Err(GetOrderError::NotFound(format!(
+				"Order not found: {}",
+				order_id
+			)))
+		}
+		Err(e) => {
+			// Other storage error
+			Err(GetOrderError::Internal(format!("Storage error: {}", e)))
+		}
+	}
+}
+
+/// Validates the order ID format.
+fn validate_order_id(order_id: &str) -> Result<(), GetOrderError> {
+	// Check if it's a valid UUID format
+	if Uuid::parse_str(order_id).is_err() {
+		return Err(GetOrderError::InvalidId(format!(
+			"Order ID must be a valid UUID: {}",
+			order_id
+		)));
+	}
+
+	Ok(())
+}
+
+/// Converts a storage Order to an API OrderResponse.
+async fn convert_order_to_response(
+	order: solver_types::Order,
+) -> Result<OrderResponse, GetOrderError> {
+	// Extract data from the order's JSON data field
+	// Return errors instead of defaulting to placeholder values
+
+	let input_amount = order.data.get("inputAmount").ok_or_else(|| {
+		GetOrderError::Internal("Missing inputAmount field in order data".to_string())
+	})?;
+	let input_amount = serde_json::from_value::<AssetAmount>(input_amount.clone())
+		.map_err(|e| GetOrderError::Internal(format!("Invalid inputAmount format: {}", e)))?;
+
+	let output_amount = order.data.get("outputAmount").ok_or_else(|| {
+		GetOrderError::Internal("Missing outputAmount field in order data".to_string())
+	})?;
+	let output_amount = serde_json::from_value::<AssetAmount>(output_amount.clone())
+		.map_err(|e| GetOrderError::Internal(format!("Invalid outputAmount format: {}", e)))?;
+
+	let settlement_type = order.data.get("settlementType").ok_or_else(|| {
+		GetOrderError::Internal("Missing settlementType field in order data".to_string())
+	})?;
+	let settlement_type = serde_json::from_value::<SettlementType>(settlement_type.clone())
+		.map_err(|e| GetOrderError::Internal(format!("Invalid settlementType format: {}", e)))?;
+
+	let settlement_data = order
+		.data
+		.get("settlementData")
+		.cloned()
+		.unwrap_or_else(|| serde_json::json!({}));
+
+	let status = order
+		.data
+		.get("status")
+		.and_then(|v| serde_json::from_value::<OrderStatus>(v.clone()).ok())
+		.ok_or_else(|| {
+			GetOrderError::Internal("Missing or invalid status field in order data".to_string())
+		})?;
```

Comment:
You will need to map this against `SolverEvent`. I'm pretty sure `OrderStatus` has a different format then the data you'll receive here.